generators/python/math.js

```diff
@@ -157,67 +157,71 @@ Blockly.Python['math_constant'] = function(block) {
-         '  if not isinstance(n, Number):',
-         '    try:',
-         '      n = float(n)',
-         '    except:',
-         '      return False',
-         '  if n == 2 or n == 3:',
-         '    return True',
-         '  # False if n is negative, is 1, or not whole,' +
-             ' or if n is divisible by 2 or 3.',
-         '  if n <= 1 or n % 1 != 0 or n % 2 == 0 or n % 3 == 0:',
-         '    return False',
-         '  # Check all the numbers of form 6k +/- 1, up to sqrt(n).',
-         '  for x in range(6, int(math.sqrt(n)) + 2, 6):',
-         '    if n % (x - 1) == 0 or n % (x + 1) == 0:',
-         '      return False',
-         '  return True']);
-    code = functionName + '(' + number_to_check + ')';
-    return [code, Blockly.Python.ORDER_FUNCTION_CALL];
-  }
-  switch (dropdown_property) {
-    case 'EVEN':
-      code = number_to_check + ' % 2 == 0';
-      break;
-    case 'ODD':
-      code = number_to_check + ' % 2 == 1';
-      break;
-    case 'WHOLE':
-      code = number_to_check + ' % 1 == 0';
-      break;
-    case 'POSITIVE':
-      code = number_to_check + ' > 0';
-      break;
-    case 'NEGATIVE':
-      code = number_to_check + ' < 0';
-      break;
-    case 'DIVISIBLE_BY':
-      var divisor = Blockly.Python.valueToCode(block, 'DIVISOR',
-          Blockly.Python.ORDER_MULTIPLICATIVE);
-      // If 'divisor' is some code that evals to 0, Python will raise an error.
-      if (!divisor || divisor == '0') {
-        return ['False', Blockly.Python.ORDER_ATOMIC];
-      }
-      code = number_to_check + ' % ' + divisor + ' == 0';
-      break;
-  }
-  return [code, Blockly.Python.ORDER_RELATIONAL];
+    const functionName = Blockly.Python.provideFunction_(
+      'math_isPrime',
+      ['def ' + Blockly.Python.FUNCTION_NAME_PLACEHOLDER_ + '(n):',
+       '  # https://en.wikipedia.org/wiki/Primality_test#Naive_methods',
+       '  # If n is not a number but a string, try parsing it.',
+       '  if not isinstance(n, Number):',
+       '    try:',
+       '      n = float(n)',
+       '    except:',
+       '      return False',
+       '  if n == 2 or n == 3:',
+       '    return True',
+       '  # False if n is negative, is 1, or not whole,' +
+       ' or if n is divisible by 2 or 3.',
+       '  if n <= 1 or n % 1 != 0 or n % 2 == 0 or n % 3 == 0:',
+       '    return False',
+       '  # Check all the numbers of form 6k +/- 1, up to sqrt(n).',
+       '  for x in range(6, int(math.sqrt(n)) + 2, 6):',
+       '    if n % (x - 1) == 0 or n % (x + 1) == 0:',
+       '      return False',
+       '  return True']);
+       code = functionName + '(' + numberToCheck + ')';
+  } else if (dropdownProperty === 'DIVISIBLE_BY') {
+    const divisor = Blockly.Python.valueToCode(block, 'DIVISOR',
+    Blockly.Python.ORDER_MULTIPLICATIVE) || '0';
+    // If 'divisor' is some code that evals to 0, Python will raise an error.
+    if (!divisor || divisor === '0') {
```

Comment:
`!divisor` is not needed, as it will always be `false`.
```suggestion
    if (divisor === '0') {
```